src/nmodl/nocpout.cpp

```diff
@@ -847,20 +876,33 @@ static const char *_mechanism[] = {\n\
     }
 
     Lappendstr(defs_list,
-               "\n\
-extern Prop* need_memb(Symbol*);\n\n\
-static void nrn_alloc(Prop* _prop) {\n\
-	Prop *prop_ion;\n\
-	double *_p; Datum *_ppvar;\n\
-");
+               "\n"
+               "extern Prop* need_memb(Symbol*);\n"
+               "static void nrn_alloc(Prop* _prop) {\n"
+               "  Prop *prop_ion{};\n"
+               "  Datum *_ppvar{};\n");
     if (point_process) {
         Lappendstr(defs_list,
-                   " if (nrn_point_prop_) {\n\
-	_prop->_alloc_seq = nrn_point_prop_->_alloc_seq;\n\
-	_p = nrn_point_prop_->param;\n\
-	_ppvar = nrn_point_prop_->dparam;\n }else{\n");
+                   " if (nrn_point_prop_) {\n"
+                   "    _nrn_mechanism_access_alloc_seq(_prop) = "
+                   "_nrn_mechanism_access_alloc_seq(nrn_point_prop_);\n"
+                   "    _ppvar = _nrn_mechanism_access_dparam(nrn_point_prop_);\n"
+                   "  } else {\n");
     }
-    Sprintf(buf, "	_p = nrn_prop_data_alloc(_mechtype, %d, _prop);\n", parraycount);
+    // need to fill _prop->dparam before calling _nrn_mechanism_cache_range(Prop*)
+    if (ppvar_cnt) {
+        Sprintf(buf, "  _ppvar = nrn_prop_datum_alloc(_mechtype, %d, _prop);\n", ppvar_cnt);
+        Lappendstr(defs_list, buf);
+        Lappendstr(defs_list, "   _nrn_mechanism_access_dparam(_prop) = _ppvar;\n");
```

Comment:
For the `assert(_nrn_mechanism_get_num_vars(_prop) == 25);`, I think one of these is "replacing" `_p = nrn_prop_data_alloc(_mechtype, 25, _prop);` and the other is replacing `_prop->param_size = 25;` -- so I think the number of literal 25s is unchanged :)

Probably obviously, these `assert`s were me being defensive and they could probably be removed at this stage.

For the `_nrn_mechanism_access_dparam(_prop) = _ppvar;` part, I think the duplication makes more sense in the context of a `POINT_PROCESS` mechanism (no doubt it could be avoided if needed).